File: online-judge-solutions/Codeforces/439A.cpp
//https://codeforces.com/problemset/problem/439/A/
//Lang		:	GNU C++14
//Time		:	15 ms
//Memory	:	0 KB
#include<iostream>
using namespace std;
int main(){
	int n,d,x,s=0;cin>>n>>d;
	for(int i=0;i<n;i++){
		cin>>x;
		s+=x;
	}
	s+=(n-1)*10;
	cout<<(d>=s?(((d>=s)?d-s:0)/5+(n-1)*2):-1)<<endl;
	return 0;
}

File: online-judge-solutions/Codeforces/439A.py
#https://codeforces.com/problemset/problem/439/A
#Lang		:	Python 3.8
#Time		:	46 ms
#Memory         :	0 KB
n,d=map(int,input().split())
s=sum(map(int,input().split()))
s+=(n-1)*10
print(int(int(d-s if d>=s else 0)/5+(n-1)*2) if d>=s else -1)
